ovs/lib/scheduledtask.py

```diff
@@ -195,186 +199,209 @@ def make_timestamp(offset):
-            vdisk = VDisk(vdisk_guid)
+        def _execute_scrub_work(queue, vpool, scrub_info):
+            client = None
+            storagerouter = scrub_info['storage_router']
+            scrub_directory = '{0}/scrub_work_{1}_{2}'.format(scrub_info['scrub_path'], vpool.name, storagerouter.name)
+            scrub_config_key = 'ovs/vpools/{0}/proxies/scrub/scrub_config_{1}'.format(vpool.guid, storagerouter.guid)
+            backend_config_key = 'ovs/vpools/{0}/proxies/scrub/backend_config_{1}'.format(vpool.guid, storagerouter.guid)
+            alba_proxy_service = 'ovs-albaproxy_{0}_{1}_scrub'.format(vpool.name, storagerouter.name)
+
+            # Deploy a proxy
             try:
-                # Load the vDisk's StorageDriver
-                ScheduledTaskController._logger.info('Execute Scrub - Virtual disk {0} - {1} - Started'.format(vdisk.guid, vdisk.name))
-                vdisk.invalidate_dynamics(['storagedriver_id'])
-                if vdisk.storagedriver_id not in storagedrivers:
-                    storagedrivers[vdisk.storagedriver_id] = StorageDriverList.get_by_storagedriver_id(vdisk.storagedriver_id)
-                storagedriver = storagedrivers[vdisk.storagedriver_id]
-
-                # Load the vDisk's MDS configuration
-                configs = _verify_mds_config(current_vdisk=vdisk)
-
-                # Check MDS master is local. Trigger MDS handover if necessary
-                if configs[0].get('ip') != storagedriver.storagerouter.ip:
-                    ScheduledTaskController._logger.debug('Execute Scrub - Virtual disk {0} - {1} - MDS master is not local, trigger handover'.format(vdisk.guid, vdisk.name))
-                    MDSServiceController.ensure_safety(vdisk)
-                    configs = _verify_mds_config(current_vdisk=vdisk)
-                    if configs[0].get('ip') != storagedriver.storagerouter.ip:
-                        skipped += 1
-                        ScheduledTaskController._logger.info('Execute Scrub - Virtual disk {0} - {1} - Skipping because master MDS still not local'.format(vdisk.guid, vdisk.name))
-                        continue
-                with vdisk.storagedriver_client.make_locked_client(str(vdisk.volume_id)) as locked_client:
-                    ScheduledTaskController._logger.info('Execute Scrub - Virtual disk {0} - {1} - Retrieve and apply scrub work'.format(vdisk.guid, vdisk.name))
-                    work_units = locked_client.get_scrubbing_workunits()
-                    for work_unit in work_units:
-                        scrubbing_result = locked_client.scrub(work_unit, scrub_location, log_sinks=[sink])
-                        locked_client.apply_scrubbing_result(scrubbing_result)
-                    if work_units:
-                        ScheduledTaskController._logger.info('Execute Scrub - Virtual disk {0} - {1} - Scrub successfully applied'.format(vdisk.guid, vdisk.name))
+                with file_mutex(name='ovs_albaproxy_scrub', wait=60):
```

Comment:
Let's increase that to ```300```, since it should possibly cover the time it takes to setup almost all proxies.